### PR TITLE
Added Prime factor algorithm in Java

### LINKED ISSUE #505: Prime Factor in Java
#### This issue is about [Algo/DS Name](link to resource for the Algo/DS)

-   [x] Issue Title is in form `[Algo/DS Name] in [Language]` (ex:- BubbleSort in Java)

-   [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate

-   [x] Assign this issue to me (I want to work on this)

-   [x] I ensure that I am not already assigned to 2 or more issues
  
-   [x] I will send my PR only after I'm assigned to this issue by any maintainer

This is an open issue for the Prime factor algorithm implementation in Java. I want to contribute, Please assign it to me.

### PR DESCRIPTION
**Pull request by JayeshJanyani**
#505 

**Fixes #505 **

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [x] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [x] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [x] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [x] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [x] This pull request shall only be merged once the build passes.

-   [x] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-prfac/java/PrimeFactor.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+public class PrimeFactor{
+
+    public static void primeFactors(int n){
```
Comment (RavinduSachintha) [inline algorithms/ar-prfac/java/PrimeFactor.java:3]: Codacy found an issue: [Avoid reassigning parameters such as 'n'](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6327890)

Comment (JayeshJanyani) [reply algorithms/ar-prfac/java/PrimeFactor.java:3]: Okay, I will make corresponding changes. 
Do I have to create a new PR?

File: algorithms/ar-prfac/java/PrimeFactor.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+public class PrimeFactor{
+
+    public static void primeFactors(int number){
```
Comment (RavinduSachintha) [inline algorithms/ar-prfac/java/PrimeFactor.java:3]: Codacy found an issue: [Avoid reassigning parameters such as 'number'](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6327890)

**Review by RavinduSachintha — APPROVED** [review]

**Comment by RavinduSachintha** [comment]
@all-contributors please add @JayeshJanyani for code and doc

**Comment by RavinduSachintha** [comment]
@JayeshJanyani good job 👍 keep it up.
And make sure to star this repo 😉

**Pull request merged**